wagtail/wagtailcore/blocks/field_block.py

```diff
@@ -383,18 +384,22 @@ def clean(self, value):
 
 
 class PageChooserBlock(ChooserBlock):
+    from wagtail.wagtailcore.models import Page
+    page_model = Page
+
     def __init__(self, can_choose_root=False, **kwargs):
         self.can_choose_root = can_choose_root
         super(PageChooserBlock, self).__init__(**kwargs)
 
     @cached_property
     def target_model(self):
-        from wagtail.wagtailcore.models import Page  # TODO: allow limiting to specific page types
-        return Page
+        return self.page_model
```

Comment:
Might be slightly simpler to set ``page_model = None`` at the top of this class and do ``return self.page_model or Page``. That way you don't need to move the import either